crates/pgt_workspace/src/workspace/server/sql_function.rs

```diff
@@ -1,56 +1,80 @@
-use std::sync::Arc;
-
-use dashmap::DashMap;
 use pgt_text_size::TextRange;
 
-use super::statement_identifier::StatementId;
+#[derive(Debug, Clone)]
+pub struct ArgType {
+    pub schema: Option<String>,
+    pub name: String,
+    pub is_array: bool,
+}
 
 #[derive(Debug, Clone)]
-pub struct SQLFunctionBody {
-    pub range: TextRange,
-    pub body: String,
+pub struct SQLFunctionArgs {
+    pub name: Option<String>,
+    pub type_: ArgType,
 }
 
-pub struct SQLFunctionBodyStore {
-    db: DashMap<StatementId, Option<Arc<SQLFunctionBody>>>,
+#[derive(Debug, Clone)]
+pub struct SQLFunctionSignature {
+    pub name: (Option<String>, String),
```

Comment:
nit: i think it would be more readable if it had a `name` and a `schema` property